crates/driver/src/tests/setup/mod.rs

```diff
@@ -777,35 +852,85 @@ impl Setup {
         // Create the necessary components for testing.
         let blockchain = Blockchain::new(blockchain::Config {
             pools,
-            trader_address,
-            trader_secret_key,
+            main_trader_secret_key: trader_secret_key,
+            trader_secret_keys: self
+                .solvers
+                .iter()
+                .filter_map(|solver| {
+                    if solver.get_configure_for_jit_orders() {
+                        Some(*solver.private_key.clone())
+                    } else {
+                        None
+                    }
+                })
+                .collect::<Vec<_>>(),
             solvers: self.solvers.clone(),
             settlement_address: self.settlement_address,
             rpc_args: self.rpc_args,
         })
         .await;
         let mut solutions = Vec::new();
+        let mut specific_solutions: HashMap<String, Vec<blockchain::Solution>> = HashMap::new();
         for solution in self.solutions {
             let orders = solution
                 .orders
                 .iter()
-                .map(|solution_order| orders.iter().find(|o| o.name == *solution_order).unwrap());
-            solutions.push(blockchain.fulfill(orders, &solution).await);
+                .flat_map(|solution_order| orders.iter().filter(|o| o.name == *solution_order))
+                .cloned()
+                .collect::<Vec<_>>();
+            let solution_fulfillment = blockchain.fulfill(orders.iter(), &solution, false).await;
+            solutions.push(solution_fulfillment.clone());
+
+            // We want to extend the solution trades with the JIT-orders only for those
+            // solvers which are configured to send the JIT-order
+            let jit_orders = solution
+                .orders
+                .iter()
+                .flat_map(|solution_order| {
+                    jit_orders
+                        .iter()
+                        .filter(|o| o.order.name == *solution_order)
+                })
+                .fold(
+                    HashMap::new(),
+                    |mut acc: HashMap<String, Vec<Order>>, jit_order| {
+                        acc.entry(jit_order.solver.to_string())
+                            .or_default()
+                            .push(jit_order.order.clone());
+                        acc
+                    },
+                )
+                .into_iter()
+                .collect::<HashMap<_, _>>();
```

Comment:
Right after collecting, there is a loop. Probably, for-each can be used here instead.